lib/rex/post/meterpreter/extensions/powershell/powershell.rb

```diff
@@ -0,0 +1,59 @@
+# -*- coding: binary -*-
+
+require 'rex/post/meterpreter/extensions/powershell/tlv'
+
+module Rex
+module Post
+module Meterpreter
+module Extensions
+module Powershell
+
+###
+#
+# This meterpreter extensions a privilege escalation interface that is capable
+# of doing things like dumping password hashes and performing local
+# exploitation.
+#
+###
+class Powershell < Extension
+
+
+  def initialize(client)
+    super(client, 'powershell')
+
+    client.register_extension_aliases(
+      [
+        {
+          'name' => 'powershell',
+          'ext'  => self
+        },
+      ])
+  end
+
+
+  def execute_string(opts={})
+    return nil unless opts[:code]
+
+    request = Packet.create_request('powershell_execute')
+    request.add_tlv(TLV_TYPE_POWERSHELL_CODE, opts[:code])
```

Comment:
Do we want to use native TLV zlib compression or the Gzip implementation in Rex::Powershell for efficient transport? I can implement this in the Post code by passhtrough method which will check compression/encoding and determine whether to compress the contents of opts[:code] prior to dispatching the result to execute_string.